pkg/jobstore/watcher.go

```diff
@@ -0,0 +1,114 @@
+package jobstore
+
+import "time"
+
+type StoreWatcherType int
+
+const (
+	JobWatcher StoreWatcherType = 1 << iota
+	ExecutionWatcher
+	EvaluationWatcher
+)
+
+func (s StoreWatcherType) String() string {
+	switch s {
+	case JobWatcher:
+		return "JobWatcher"
+	case ExecutionWatcher:
+		return "ExecutionWatcher"
+	case EvaluationWatcher:
+		return "EvaluationWatcher"
+	}
+	return "Unknown type"
+}
+
+type StoreEventType int
+
+const (
+	CreateEvent StoreEventType = 1 << iota
+	UpdateEvent
+	DeleteEvent
+)
+
+func (s StoreEventType) String() string {
+	switch s {
+	case CreateEvent:
+		return "CreateEvent"
+	case UpdateEvent:
+		return "UpdateEvent"
+	case DeleteEvent:
+		return "DeleteEvent"
+	}
+	return "Unknown event"
+}
+
+const DefaultWatchChannelSize = 64
+
+// WatchEvent is the message passed through the watcher whenever a
+// specific event occurs on a specific type, as requested when creating
+// the watcher.
+type WatchEvent struct {
+	Kind      StoreWatcherType
+	Event     StoreEventType
+	Object    []byte
+	Timestamp int64
+}
+
+func NewWatchEvent(kind StoreWatcherType, event StoreEventType, object []byte) WatchEvent {
+	return WatchEvent{
+		Kind:      kind,
+		Event:     event,
+		Object:    append([]byte(nil), object...),
+		Timestamp: time.Now().Unix(),
```

Comment:
Timestamp is the time the event was published and not necessary when the change has happened. It can be confusing and I am concerned people will use it wrongly. My recommendation is to remove the field and publish the actual event time, which I understand can be challenging